kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/SnapshotQueryContext.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.metrics;
+
+import io.delta.kernel.metrics.SnapshotReport;
+import java.util.Optional;
+
+/** Stores the context for a given Snapshot query. Used to generate a {@link SnapshotReport} */
```

Comment:
This class is named `SnapshotQueryContext` ... and you say that it "stores the context" ... but can you define a bit what "context" means? You don't need to list all of the information it contains .. but at high level, what does context mean?

```
Stores the context for a given Snapshot query. Used to generate a {@link SnapshotReport}.

Here, "context" means the state and information pertaining to when a Snapshot was constructed, for which table, and its associated metrics
```

Reply:
Can you also talk a bit about the expected lifecycle of this class. What is it coupled to?

Reply:
Updated the class docs